tuf/data/types_test.go

```diff
@@ -53,3 +53,108 @@ func TestSignatureUnmarshalJSON(t *testing.T) {
 	// Check that the method string is lowercased
 	assert.Equal(t, sig.Method.String(), "rsa")
 }
+
+func TestCheckHashes(t *testing.T) {
+	var err error
+	raw := []byte("Bumblebee")
+
+	// Since only provide an un-supported hash algorithm here,
+	// it should be considered as fail.
+	unSupported := make(Hashes)
+	unSupported["Arthas"] = []byte("is past away.")
+	err = CheckHashes(raw, unSupported)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "at least one supported hash needed")
+
+	// Expected to fail since there is no checksum at all.
+	hashes := make(Hashes)
+	err = CheckHashes(raw, hashes)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "at least one supported hash needed")
+
+	// The most standard one.
+	hashes["sha256"], err = hex.DecodeString("d13e2b60d74c2e6f4f449b5e536814edf9a4827f5a9f4f957fc92e77609b9c92")
+	assert.NoError(t, err)
+	hashes["sha512"], err = hex.DecodeString("f2330f50d0f3ee56cf0d7f66aad8205e0cb9972c323208ffaa914ef7b3c240ae4774b5bbd1db2ce226ee967cfa9058173a853944f9b44e2e08abca385e2b7ed4")
+	assert.NoError(t, err)
+	err = CheckHashes(raw, hashes)
+	assert.NoError(t, err)
+
+	// Expected as succeed since there are already supported hash here,
+	// just ignore the unsupported one.
+	hashes["Saar"] = []byte("survives again in CTM.")
+	err = CheckHashes(raw, hashes)
+	assert.NoError(t, err)
+
+	only256 := make(Hashes)
+	only256["sha256"], err = hex.DecodeString("d13e2b60d74c2e6f4f449b5e536814edf9a4827f5a9f4f957fc92e77609b9c92")
+	assert.NoError(t, err)
+	err = CheckHashes(raw, only256)
+	assert.NoError(t, err)
+
+	only512 := make(Hashes)
+	only512["sha512"], err = hex.DecodeString("f2330f50d0f3ee56cf0d7f66aad8205e0cb9972c323208ffaa914ef7b3c240ae4774b5bbd1db2ce226ee967cfa9058173a853944f9b44e2e08abca385e2b7ed4")
+	assert.NoError(t, err)
+	err = CheckHashes(raw, only512)
+	assert.NoError(t, err)
+
+	// Expected to fail due to the failure of sha256
+	malicious256 := make(Hashes)
+	malicious256["sha256"] = []byte("malicious data")
+	err = CheckHashes(raw, malicious256)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "checksum mismatched")
+
+	// Expected to fail due to the failure of sha512
+	malicious512 := make(Hashes)
+	malicious512["sha512"] = []byte("malicious data")
+	err = CheckHashes(raw, malicious512)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "checksum mismatched")
+}
+
+func TestGetSupportedHashes(t *testing.T) {
+	raw := []byte("Illidan")
+
+	hashes := GetSupportedHashes(raw)
+	err := CheckHashes(raw, hashes)
+	assert.NoError(t, err)
+}
+
+func TestCheckValidHashStructures(t *testing.T) {
+	var err error
+	hashes := make(Hashes)
+
+	// Expected to fail since there is no checksum at all.
+	err = CheckValidHashStructures(hashes)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "at least one supported hash needed")
+
+	// Expected to fail even though the checksum of sha384 is valid,
+	// because we havn't provide a supported hash algorithm yet.
```

Comment:
@riyazdf `we haven't supported that hash algorithm yet` doesn't always mean to fail this test, in this case it failed because only `sha384` was provided. 

If the `sha384` comes along with a supported one, let's say `sha256`, then this case will expected as success